src/gui/qgsrichtexteditor.cpp

```diff
@@ -752,16 +752,20 @@ void QgsRichTextEditor::setText( const QString &text )
 {
   if ( text.isEmpty() )
   {
-    setPlainText( text );
+    mTextEdit->setPlainText( text );
+    mSourceEdit->clear();
     return;
   }
+
   if ( text[0] == '<' )
```

Comment:
Should we trim the text here?

Reply:
I don't think so -- maybe the user wants spaces at the sides

Reply:
Right but then we'll end up handling the string as a non-HTML one no? Shouldn't we do the check against a trimmed version (but then use the non trimmed version when assigning to the text and source edit widgets)?

Reply:
Ok, I've implemented this logic